decidim-surveys/spec/system/private_space_survey_spec.rb

```diff
@@ -28,7 +28,7 @@
   let!(:participatory_space_private_user) { create(:participatory_space_private_user, user: another_user, privatable_to: participatory_space_private) }
 
   let!(:questionnaire) { create(:questionnaire, title:, description:) }
-  let!(:survey) { create(:survey, component:, questionnaire:) }
+  let!(:survey) { create(:survey, component:, published_at: Time.current, questionnaire:) }
```

Comment:
```suggestion
  let!(:survey) { create(:survey, :published, :allow_answers, component:, questionnaire:) }
```
I'd create a new trait in the factories to add a `allow_answers" helper. This way you can avoid the "before" block this:
`survey.update!(allow_answers: true)`

Reply:
Done